### PR TITLE
[node] Add Buffer.alloc methods to node v4

### PR DESCRIPTION
**Pull request by ariaviran**
Add `Buffer.alloc`, `Buffer.allocUnsafe` and `Buffer.allocUnsafeSlow`
class methods that were backported to node v4 (starting with v4.5).

In addition add tests to all node versions for these class methods and
fix tslint configuration to pass the tests

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V4.md#2016-08-15-version-450-argon-lts-thealphanerd
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/node/tslint.json [hunk 1]
```diff
@@ -21,6 +21,7 @@
     "prefer-const": false,
     "prefer-method-signature": false,
     "strict-export-declare-modifiers": false,
-    "unified-signatures": false
+    "unified-signatures": false,
+    "void-return": false
```
Comment (Flarna) [inline types/node/tslint.json]: Why is it needed to disable the `void-return` lint warning?

Comment (ariaviran) [reply types/node/tslint.json]: The typings are full of `void` and `Promise<void>` return types, I honestly don't know how it passed until now

**Review by Flarna — APPROVED** [review]

File: types/node/index.d.ts [hunk 2]
```diff
@@ -104,8 +104,8 @@ declare namespace setImmediate {
 declare function clearImmediate(immediateId: any): void;
 
 // TODO: change to `type NodeRequireFunction = (id: string) => any;` in next mayor version.
-/* tslint:disable:callable-types */
 interface NodeRequireFunction {
+    /* tslint:disable-next-line:callable-types */
```
Comment (eps1lon) [inline types/node/index.d.ts]: This is a duplicate of #21768

File: types/node/tslint.json (hunk 1 above)
Comment (eps1lon) [reply types/node/tslint.json]: This is a false positive. Void is nowhere used as a param and only as a return type which is the kind of behavior the [rule](https://github.com/Microsoft/dtslint/blob/master/docs/void-return.md) enforces

**Comment by ariaviran** [comment]
@eps1lon Thank you for your comments, I've fixed them

**Pull request merged**